lib/Dialect/CGGI/Transforms/BooleanVectorizer.cpp

```diff
@@ -119,27 +120,28 @@ bool tryBoolVectorizeBlock(Block *block, MLIRContext &context) {
           {static_cast<int64_t>(bucket.size())}, elementType);
 
       SmallVector<Value, 4> vectorizedOperands;
-      SmallVector<StringAttr, 4> vectorizedGateOperands;
+      SmallVector<CGGIBoolGateEnumAttr, 4> vectorizedGateOperands;
 
       for (auto *op : bucket) {
-        std::string str;
-        if (isa<cggi::AndOp>(op)) {
-          str = "and";
-        } else if (isa<cggi::NandOp>(op)) {
-          str = "nand";
-        } else if (isa<cggi::XorOp>(op)) {
-          str = "xor";
-        } else if (isa<cggi::XNorOp>(op)) {
-          str = "xnor";
-        } else if (isa<cggi::OrOp>(op)) {
-          str = "or";
-        } else if (isa<cggi::NorOp>(op)) {
-          str = "nor";
-        } else {
-          LLVM_DEBUG(llvm::dbgs()
-                     << "Try to parse boolean operation that does not exist.");
-        }
-        vectorizedGateOperands.push_back(StringAttr::get(&context, str));
+        auto gateStr =
+            llvm::TypeSwitch<Operation *, FailureOr<CGGIBoolGateEnum>>(op)
+                .Case<cggi::AndOp>(
+                    [](AndOp op) { return CGGIBoolGateEnum::bg_and; })
+                .Case<cggi::NandOp>(
+                    [](NandOp op) { return CGGIBoolGateEnum::bg_nand; })
+                .Case<cggi::XorOp>(
+                    [](XorOp op) { return CGGIBoolGateEnum::bg_xor; })
+                .Case<cggi::XNorOp>(
+                    [](XNorOp op) { return CGGIBoolGateEnum::bg_xnor; })
+                .Case<cggi::OrOp>(
+                    [](OrOp op) { return CGGIBoolGateEnum::bg_or; })
+                .Case<cggi::NorOp>(
+                    [](NorOp op) { return CGGIBoolGateEnum::bg_nor; })
+                .Default(
+                    [](Operation *) { return FailureOr<CGGIBoolGateEnum>(); });
+
```

Comment:
```suggestion
assert(succeeded(op))
```

or return false if failed because the pass would fail otherwise